src/overlays/actors/ovl_Bg_Jya_Megami/z_bg_jya_megami.h

```diff
@@ -6,9 +6,34 @@
 
 struct BgJyaMegami;
 
+typedef void (*BgJyaMegamiActionFunc)(struct BgJyaMegami*, GlobalContext*);
+
+typedef struct BgJyaMegami_PieceInit {
+    /* 0x00 */ Vec3f unk_00;
+    /* 0x0C */ f32 velX;
+    /* 0x10 */ s16 rotVelX;
+    /* 0x12 */ s16 rotVelY;
+    /* 0x14 */ s16 delay;
+} BgJyaMegami_PieceInit; // size = 0x18
+
+typedef struct BgJyaMegami_Piece {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ Vec3f vel;
+    /* 0x18 */ s16 rotVelX;
+    /* 0x1A */ s16 rotVelY;
+} BgJyaMegami_Piece; // size = 0x1C
+
+static BgJyaMegami_PieceInit sPiecesInit[13];
```

Comment:
I dont know if including BgJyaMegami_PieceInit and the sPiecesInit declaration in the header is worth it for the sake of using array count. but i want others opinions as well

Reply:
seems worth it to me for the sake of easy modding, but understandable either way

Reply:
I'm for it in theory, but I wish we had a more consistent approach to this

Reply:
Doing it this way is very confusing and unfortunate imo. How about having a define for the piece count, used as the size of both arrays?